aiida_mlip/parsers/train_parser.py

```diff
@@ -0,0 +1,211 @@
+
+        self._save_models(model_output, compiled_model_output)
+        self._parse_results(result_name)
+        self._save_folders(remote_dirs)
+
+        return ExitCode(0)
+
+    def _get_remote_dirs(self, mlip_dict: dict) -> dict:
+        """
+        Get the remote directories based on mlip config file.
+
+        Parameters
+        ----------
+        mlip_dict : dict
+            Dictionary containing mlip config file.
+
+        Returns
+        -------
+        dict
+            Dictionary of remote directories.
+        """
+        rem_dir = Path(self.node.get_remote_workdir())
+        return {
+            typ: rem_dir / mlip_dict.get(f"{typ}_dir", default)
+            for typ, default in (
+                ("log", "logs"),
+                ("checkpoint", "checkpoints"),
+                ("results", "results"),
+                ("model", ""),
+            )
+        }
+
+    def _validate_retrieved_files(self, output_filename: str, model_name: str) -> bool:
+        """
+        Validate that the expected files have been retrieved.
+
+        Parameters
+        ----------
+        output_filename : str
+            The expected output filename.
+        model_name : str
+            The name of the model as found in the config file key `name`.
+
+        Returns
+        -------
+        bool
+            True if the expected files are retrieved, False otherwise.
+        """
+        files_retrieved = self.retrieved.list_object_names()
+        files_expected = {output_filename, f"{model_name}.model"}
+
+        if not files_expected.issubset(files_retrieved):
+            self.logger.error(
+                f"Found files '{files_retrieved}', expected to find '{files_expected}'"
+            )
+            return False
+        return True
+
+    def _save_models(self, model_output: Path, compiled_model_output: Path) -> None:
+        """
+        Save model and compiled model as outputs.
+
+        Parameters
+        ----------
+        model_output : Path
+            Path to the model output file.
+        compiled_model_output : Path
+            Path to the compiled model output file.
+        """
+        architecture = "mace_mp"
+        model = ModelData.local_file(model_output, architecture=architecture)
+        compiled_model = ModelData.local_file(
+            compiled_model_output, architecture=architecture
+        )
+
+        self.out("model", model)
+        self.out("compiled_model", compiled_model)
+
+    def _parse_results(self, result_name: Path) -> None:
+        """
+        Parse the results file and store the results dictionary.
+
+        Parameters
+        ----------
+        result_name : Path
+            Path to the result file.
+        """
+        with open(result_name, encoding="utf-8") as file:
+            last_dict_str = None
+            for line in file:
+                try:
+                    last_dict_str = ast.literal_eval(line.strip())
```

Comment:
`ast.literal_eval` raises big concerns. What is this trying to do? What is the format of this file? If it's JSON like (Python `dict` like) should be using stdlib json parser.